spec/coordinates_spec.rb

```diff
@@ -0,0 +1,87 @@
+require "spec_helper"
+require_relative "../lib/coordinates.rb"
+require_relative "../lib/grid.rb"
+
+RSpec.describe Coordinates do
+  describe "#valid?" do
```

Comment:
According to the code, there are only two tests that you need to build:
1. The grid object return false from `invalid_coordinate?`
2. The grid object returns true from `invalid_coordinate?`

The grid object is where all the logic happens so the tests should not be overly familiar with the logic the grid is using to determine that.

Reply:
Right! Of course!